spec/system/consumer/shopping/variant_overrides_spec.rb

```diff
@@ -18,21 +18,33 @@
   let(:outgoing_exchange) { oc.exchanges.outgoing.first }
   let(:sm) { hub.shipping_methods.first }
   let(:pm) { hub.payment_methods.first }
-  let(:product1) { create(:simple_product, supplier: producer) }
-  let(:product2) { create(:simple_product, supplier: producer) }
-  let(:product3) { create(:simple_product, supplier: producer, on_demand: true) }
-  let(:product4) { create(:simple_product, supplier: producer) }
-  let(:product1_variant1) { create(:variant, product: product1, price: 11.11, unit_value: 1) }
-  let(:product1_variant2) { create(:variant, product: product1, price: 22.22, unit_value: 2) }
-  let(:product2_variant1) { create(:variant, product: product2, price: 33.33, unit_value: 3) }
-  let(:product1_variant3) { create(:variant, product: product1, price: 44.44, unit_value: 4) }
+  let(:product1) { create(:simple_product, supplier_id: producer.id) }
+  let(:product2) { create(:simple_product, supplier_id: producer.id) }
+  let(:product3) { create(:simple_product, supplier_id: producer.id, on_demand: true) }
+  let(:product4) { create(:simple_product, supplier_id: producer.id) }
+  let(:product1_variant1) {
+    create(:variant, product: product1, price: 11.11, unit_value: 1, supplier: producer)
+  }
+  let(:product1_variant2) {
+    create(:variant, product: product1, price: 22.22, unit_value: 2, supplier: producer)
+  }
+  let(:product2_variant1) {
+    create(:variant, product: product2, price: 33.33, unit_value: 3, supplier: producer)
+  }
+  let(:product1_variant3) {
+    create(:variant, product: product1, price: 44.44, unit_value: 4, supplier: producer)
```

Comment:
It would be interesting to spec what happens with a different supplier on the variant. But I'm not confident enough to define what should happen, so let's leave it undefined until it gets raised as an issue.. ;)